cloud/keystone.go

```diff
@@ -228,6 +232,15 @@ func isTokenScopedToAdminProject(result *v3tokens.GetResult) bool {
 	return (err == nil) && s.IsAdminProject
 }
 
+func (client *keystoneV3Client) validateRolesForAdmin(roleIDs []string) error {
+	if len(roleIDs) != 1 || roleIDs[0] != schema.AdminRole {
+		client.updateCounter(1, "error.token.mismatched_role_admin")
```

Comment:
Is it useful to have metric for such a specific event?

Reply:
It doesn't hurt. There are already multiple different metrics for different failure cases.